ci/scripts/cpp_build.sh

```diff
@@ -283,9 +283,10 @@ else
 fi
 
 if [ "${ARROW_USE_MESON:-OFF}" = "ON" ]; then
-  time meson install
+  time meson compile -j ${ARROW_BUILD_PARALLEL:-$[${n_jobs} + 1]}
+  meson install
 else
-  export CMAKE_BUILD_PARALLEL_LEVEL=${CMAKE_BUILD_PARALLEL_LEVEL:-$[${n_jobs} + 1]}
+  export CMAKE_BUILD_PARALLEL_LEVEL=${ARROW_BUILD_PARALLEL:-$[${n_jobs} + 1]}
```

Comment:
Could you use `ARROW_BUILD_PARALLEL` only if `CMAKE_BUILD_PARALLEL_LEVEL`? (If both of `CMAKE_BUILD_PARALLEL_LEVEL` and `ARROW_BUILD_PARALLEL`, `CMAKE_BUILD_PARALLEL_LEVEL` should be used.)

Reply:
It looks a bit more complicated but let me know if the current iteration is now inline with your expectations or not